packages/js-sdk/src/Dm3Sdk.test.ts

```diff
@@ -1,34 +1,278 @@
+                    if (
+                        normalizeEnsName(ensName) ===
+                        normalizeEnsName(bob.address) + '.addr.test'
+                    ) {
+                        return 'bob.eth';
+                    }
+                    return 'karl.eth';
+                },
+            } as unknown as ITLDResolver;
+
+            const mockConfig: Dm3SdkConfig = {
+                mainnetProvider: {} as ethers.providers.JsonRpcProvider,
+                storageApi: {
+                    addConversation: async () => {},
+                } as unknown as StorageAPI,
+                nonce: '1',
+                defaultDeliveryService: 'test.io',
+                addressEnsSubdomain: '.addr.test',
+                userEnsSubdomain: '.user.test',
+                resolverBackendUrl: 'resolver.io',
+                backendUrl: 'http://localhost:4060',
+                _tld: mockTldResolver,
+            };
+
+            const dm3 = await new Dm3Sdk(mockConfig).login({
+                profileKeys: alice.profileKeys,
+                profile: alice.signedUserProfile,
+                accountAddress: alice.address,
+            });
+
+            await dm3.conversations.addConversation('bob.eth');
+            await dm3.conversations.addConversation('karl.eth');
+            const c = dm3.conversations.list;
+            dm3;
+            console.log(c);
+            expect(c.length).toBe(2);
+            expect(c[0].contact.name).toBe('bob.eth');
+            expect(c[1].contact.name).toBe('karl.eth');
+        });
+        it('dont add duplicate conversations', async () => {
+            const mockTldResolver = {
+                resolveTLDtoAlias: async () =>
+                    `${normalizeEnsName(bob.address)}.addr.test`,
+                resolveAliasToTLD: async () => 'bob.eth',
+            } as unknown as ITLDResolver;
+
+            const mockConfig: Dm3SdkConfig = {
+                mainnetProvider: {} as ethers.providers.JsonRpcProvider,
+                storageApi: {
+                    addConversation: async () => {},
+                } as unknown as StorageAPI,
+                nonce: '1',
+                defaultDeliveryService: 'test.io',
+                addressEnsSubdomain: '.addr.test',
+                userEnsSubdomain: '.user.test',
+                resolverBackendUrl: 'resolver.io',
+                backendUrl: 'http://localhost:4060',
+                _tld: mockTldResolver,
+            };
+
+            const dm3 = await new Dm3Sdk(mockConfig).login({
+                profileKeys: alice.profileKeys,
+                profile: alice.signedUserProfile,
+                accountAddress: alice.address,
+            });
+
+            await dm3.conversations.addConversation('bob.eth');
+            await dm3.conversations.addConversation('bob.eth');
+            const c = dm3.conversations.list;
+            expect(c.length).toBe(1);
+            expect(c[0].contact.name).toBe('bob.eth');
```

Comment:
should suffice without additional variable?

```suggestion
            expect(dm3.conversations.list.length).toBe(1);
            expect(dm3.conversations.list[0].contact.name).toBe('bob.eth');
```